src/components/pastAnalysis/seriesTable.tsx

```diff
@@ -11,52 +11,55 @@ interface SeriesTableProps {
   studyInstance: StudyInstanceWithSeries
 }
 
-export const isLargestNumber = (num: number | undefined, compare1: number | undefined, compare2: number | undefined) => {
-  if (num !== undefined && compare1 !== undefined && compare2 !== undefined)
-    return num === Math.max(num, compare1, compare2);
+export const isLargestNumber = (num: number | null, numArray: number[]) => {
+  if (num !== null && numArray !== null) {
+    return num === Math.max(...numArray);
+  }
   else return false
 }
 
 const SeriesTable: React.FC<SeriesTableProps> = ({ studyInstance }) => {
   const history = useHistory();
   const { dispatch } = useContext(AppContext);
-  const { series: analysisList } = studyInstance
-  const columns = [
-    { title: (<span><br />Image<span className='classificationText'>&nbsp;</span></span>) },
-    { title: (<span>Predictions<br /><span className='classificationText'>COVID-19</span></span>) },
-    { title: (<span><br /><span className='classificationText'>Pneumonia</span></span>) },
-    { title: (<span><br /><span className='classificationText'>Normal</span></span>) },
-    { title: (<span className='classificationText'><span>Geographic<br />Severity</span></span>) },
-    { title: (<span className='classificationText'><span>Opacity<br />Extent</span></span>) },
-    { title: (<span></span>) }
-  ]
+  const { series: analysisList } = studyInstance;
+  let titles = [{ title: (<span><br />Image<span className='classificationText'>&nbsp;</span></span>) }];
+
+  for (let title of analysisList[0]?.columnNames) { // Adding the column titles for each analysis
```

Comment:
Did some digging into this with Janakitti, and it turns out that each analysisList contains the images picked from the same study.
And since you can only pick one model for each type of study currently, each analysis in the analysisList will have the same columnNames